mergo_test.go

```diff
@@ -155,7 +155,7 @@ func TestPointerStruct(t *testing.T) {
 }
 
 type embeddingStruct struct {
-	embeddedStruct
+	A embeddedStruct
```

Comment:
The field either has to be exported or the unit test needs to change:
```
			src: embeddingStruct{
				embeddedStruct{"foo"},
			},
			dst: embeddingStruct{
				embeddedStruct{""},
			},
			expected: embeddingStruct{
				embeddedStruct{"foo"},
			},
		},
```
Merging into a private field of a struct cannot should not be replaced. `hasExportedField` sees that the sub struct has exported fields which to me does not make much sense as the parent does not and that is the important part. So I am not sure that is the behavior we want to have.